java/ql/lib/semmle/code/java/security/TempFileLib.qll

```diff
@@ -0,0 +1,33 @@
+/** Provides classes to reason about temporary directory vulnerabilities. */
+
+import java
+import semmle.code.java.dataflow.ExternalFlow
+
+/**
+ * All `java.io.File::createTempFile` methods.
+ */
+class MethodFileCreateTempFile extends Method {
+  MethodFileCreateTempFile() {
+    this.getDeclaringType() instanceof TypeFile and
+    this.hasName("createTempFile")
+  }
+}
+
+/**
+ * All methods on the class `java.io.File` that create directories.
+ */
+class MethodFileCreatesDirs extends Method {
+  MethodFileCreatesDirs() {
+    getDeclaringType() instanceof TypeFile and
+    hasName(["mkdir", "mkdirs"])
+  }
+}
+
+private class TemporaryFileFlow extends SummaryModelCsv {
```

Comment:
I intentionally don't want this to be defined at a global scope, yet. I don't know what the impact of this would be on any of the path normalization (path traversal) based queries and I don't really want to deal with that rn

Reply:
If it's only imported in some queries and not in others then it will lead to a whole bunch of cache invalidation and reevaluation of major parts of the libraries, which isn't acceptable. So if these steps only needs to be active for certain queries on an opt-in basis, then they can't use the csv format and must be defined fully in QL as pairs of data-flow nodes and directly mentioned by the configurations that wants to opt-in.

Reply:
I've broken this into an independent PR to allow this change to be tested independently. Thus, if it introduces FP into other queries, that can be analyzed independently: https://github.com/github/codeql/pull/8884